src/os/filestore/FileStore.cc

```diff
@@ -237,18 +237,22 @@ int FileStore::lfn_open(const coll_t& cid,
 			const ghobject_t& oid,
 			bool create,
 			FDRef *outfd,
-                        Index *index)
+			Index *index,
+			bool directio)
 {
   assert(outfd);
   int r = 0;
   bool need_lock = true;
   int flags = O_RDWR;
+  FDCache *cachep = directio ? &fdcache_direct : &fdcache;
 
   if (create)
     flags |= O_CREAT;
   if (g_conf->filestore_odsync_write) {
     flags |= O_DSYNC;
   }
+  if (directio)
+    flags |= O_DIRECT;
```

Comment:
There's a more fundamental problem here. The kernel does not cache O_DIRECT reads, so every request will be served directly from the hard drive. This will deteriorate both read and write performance of rotating hard drives and might shorten the SSDs lifetime (this is not an issue for journals -- they are write mostly, and writes should go directly to the hard drive anyway).